### ISSUE TITLE
Mocha isn't running when installed locally...

### ISSUE THREAD

**Issue by balupton** [mochajs/mocha#291]
For some strange reason, mocha is deciding not to run when installed locally anymore... I have no clue why this is, and it only seems to happen on my iMac (works fine on my macbook).

Here's the log:

```
$ npm install mocha@0.13
npm http GET https://registry.npmjs.org/mocha
npm http 304 https://registry.npmjs.org/mocha
npm http GET https://registry.npmjs.org/commander
npm http GET https://registry.npmjs.org/growl
npm http GET https://registry.npmjs.org/jade/0.20.3
npm http GET https://registry.npmjs.org/debug
npm http 304 https://registry.npmjs.org/growl
npm http 304 https://registry.npmjs.org/debug
npm http 304 https://registry.npmjs.org/jade/0.20.3
npm http 304 https://registry.npmjs.org/commander
npm http GET https://registry.npmjs.org/mkdirp
npm http 304 https://registry.npmjs.org/mkdirp
mocha@0.13.0 ./node_modules/mocha 
├── growl@1.5.0
├── debug@0.5.0
├── commander@0.5.2
└── jade@0.20.3
$ npm install -g mocha@0.13
npm http GET https://registry.npmjs.org/mocha
npm http 304 https://registry.npmjs.org/mocha
npm http GET https://registry.npmjs.org/commander
npm http GET https://registry.npmjs.org/growl
npm http GET https://registry.npmjs.org/jade/0.20.3
npm http GET https://registry.npmjs.org/debug
npm http 304 https://registry.npmjs.org/commander
npm http 304 https://registry.npmjs.org/jade/0.20.3
npm http 304 https://registry.npmjs.org/growl
npm http 304 https://registry.npmjs.org/debug
npm http GET https://registry.npmjs.org/mkdirp
npm http 304 https://registry.npmjs.org/mkdirp
/Users/balupton/.nvm/v0.6.11/bin/mocha -> /Users/balupton/.nvm/v0.6.11/lib/node_modules/mocha/bin/mocha
/Users/balupton/.nvm/v0.6.11/bin/_mocha -> /Users/balupton/.nvm/v0.6.11/lib/node_modules/mocha/bin/_mocha
growl@1.5.0 /Users/balupton/.nvm/v0.6.11/lib/node_modules/mocha/node_modules/growl 
debug@0.5.0 /Users/balupton/.nvm/v0.6.11/lib/node_modules/mocha/node_modules/debug 
commander@0.5.2 /Users/balupton/.nvm/v0.6.11/lib/node_modules/mocha/node_modules/commander 
mkdirp@0.3.0 /Users/balupton/.nvm/v0.6.11/lib/node_modules/mocha/node_modules/jade/node_modules/mkdirp 
jade@0.20.3 /Users/balupton/.nvm/v0.6.11/lib/node_modules/mocha/node_modules/jade 
mocha@0.13.0 /Users/balupton/.nvm/v0.6.11/lib/node_modules/mocha
$ node ./node_modules/mocha/bin/mocha  --reporter spec  --ui bdd  --ignore-leaks  --growl
$ ./node_modules/mocha/bin/mocha  --reporter spec  --ui bdd  --ignore-leaks  --growl
$ mocha  --reporter spec  --ui bdd  --ignore-leaks  --growl


  versionCompare
    ✓ should detect 1.1.0 < 1.2.0 is true 
    ✓ should detect 1.2.0 > 1.1.0 is true 
    ✓ should detect 1.2.0 == 1.2.0 is true 
    ✓ should detect 1.2.0 <= 1.2.0 is true 
    ✓ should detect 1.2.0 >= 1.2.0 is true 
    ✓ should detect 1.2.0 < 1.2.1 is true 
    ✓ should detect 1.2.0 < 1.3.0 is true 
    ✓ should detect 1.1.0 >= 1.2.0 is false 
    ✓ should detect 1.2.0 <= 1.1.0 is false 
    ✓ should detect 1.2.0 <> 1.2.0 is false 
    ✓ should detect 1.2.0 > 1.2.0 is false 
    ✓ should detect 1.2.0 < 1.2.0 is false 
    ✓ should detect 1.2.0 >= 1.2.1 is false 
    ✓ should detect 1.2.0 >= 1.3.0 is false 

  packageCompare
    ◦ should run as expected: c^C$ 
$ node --version
v0.6.11
$ npm --version
1.1.1
```

Ran against the repository: https://github.com/balupton/bal-util.npm

**Comment by SherylHohman** [comment]
after `npm install mocha --save-dev`
I had the same problem: $`mocha --version` "`-bash: mocha: command not found`"

`npm cache clean` did not work for me, but adding 

`npm install -g mocha --save-dev` "fixed" the problem.

**Comment by SherylHohman** [comment]
I found this for further insight:

http://www.unknownerror.org/opensource/mochajs/mocha/q/stackoverflow/9722407/how-do-you-install-and-run-mocha-the-node-js-testing-module-getting-quotmocha-co

tldr;
- install mocha globally
OR
- Add ./node_modules/.bin to my PATH

(in the post, he First Downgraded to Node.js 0.6.12 - I'm not sure if this is necessary)

**Comment by ScottFreeCode** [comment]
As I understand it, this is standard NPM behavior.

- They say ([here](https://docs.npmjs.com/files/folders), [here](https://docs.npmjs.com/getting-started/installing-npm-packages-globally) and [here](https://docs.npmjs.com/getting-started/installing-npm-packages-locally)) that you should install a module locally if you intend to `require` it, and globally if you intend to run it on the commandline.
- The node_modules/.bin folder also comes into play for npm scripts (see [here](https://docs.npmjs.com/misc/scripts#path) and [here](https://docs.npmjs.com/cli/run-script)). So, for instance, you can add your mocha testing command to package.json's `scripts.test` property and then call [`npm test`](https://docs.npmjs.com/cli/test) to run it with local Mocha rather than global -- useful, for instance, in continuous integration.

We should double-check that it's properly documented, though (at least, that the documentation we do have for the CLI isn't misleading on this point).

**Comment by AlbertHambardzumyan** [comment]
Getting such error:
ENOENT: no such file or directory, mkdir '.....project_path\node_modules\mocha\bin\public\files'

node 6.10.1
npm 4.4.4

**Comment by ScottFreeCode** [comment]
@AlbertHambardzumyan I don't see any references to the `bin/public/files` path in Mocha's source code using GitHub search; if you can get us [a basic setup to reproduce your error](http://stackoverflow.com/help/mcve), then please open a new issue for it.

**Comment by MiroHibler** [comment]
Adding `./node_modules/.bin` to my PATH worked for me! 🚀

**Comment by danger89** [comment]
Use: `npm test` instead

**Comment by AhzamBardai** [comment]
node_modules\.bin\mocha:2
basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")
          ^^^^^^^

SyntaxError: missing ) after argument list

getting an error in the terminal when `npm test` is run
